action/envelope.go

```diff
@@ -63,6 +65,20 @@ func (elp *envelope) IntrinsicGas() (uint64, error) {
 // Action returns the action payload.
 func (elp *envelope) Action() Action { return elp.payload }
 
+// ToEthTx converts to Ethereum tx
+func (elp *envelope) ToEthTx(evmNetworkID uint32, encoding iotextypes.Encoding) (*types.Transaction, error) {
+	switch {
+	// TODO: handle dynamic fee tx and blob tx
+	case encoding == iotextypes.Encoding_IOTEX_PROTOBUF:
+		// treat native tx as EVM LegacyTx
+		fallthrough
+	case encoding == iotextypes.Encoding_ETHEREUM_EIP155 || encoding == iotextypes.Encoding_ETHEREUM_UNPROTECTED:
+		return toLegacyTx(&elp.AbstractAction, elp.Action())
```

Comment:
generate Ethereum tx from 2 parts of data:
1. common data (like nonce, gasprice, gaslimit, and blob data in upcoming EIP-4844 tx)
2. action-specific data from `elp.Action` (like transfer/execution/staking/rewarding)